gson/src/test/java/com/google/gson/internal/LinkedTreeMapTest.java

```diff
@@ -161,7 +161,7 @@ public void testJavaSerialization() throws IOException, ClassNotFoundException {
   }
 
   @SafeVarargs
-  private <T> void assertIterationOrder(Iterable<T> actual, T... expected) {
+  private final <T> void assertIterationOrder(Iterable<T> actual, T... expected) {
```

Comment:
Interestingly when changing the version to Java 7 this caused a build failure (due to [private methods not supporting it in Java 7](https://bugs.openjdk.java.net/browse/JDK-8047159)), but when the version was Java 6 this compiled fine. Most likely because `@SafeVarargs` was added in Java 7, but ([unlike `--release`](https://stackoverflow.com/a/43103038)) the `-source` and `-target` compiler flags do not prevent using newer API.